microk8s-resources/wrappers/microk8s-enable.wrapper

```diff
@@ -1,64 +1,60 @@
-#!/bin/bash
+#!/usr/bin/env python
 
+import sys
+import textwrap
+from pathlib import Path
+from subprocess import run
 
-export PATH="$SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH"
-ARCH="$($SNAP/bin/uname -m)"
-export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$SNAP/lib:$SNAP/usr/lib:$SNAP/lib/$ARCH-linux-gnu:$SNAP/usr/lib/$ARCH-linux-gnu"
-export LD_LIBRARY_PATH=$SNAP_LIBRARY_PATH:$LD_LIBRARY_PATH
+import click
 
-source $SNAP/actions/common/utils.sh
+from utils import check_clustering
 
-if [ -e ${SNAP_DATA}/var/lock/clustered.lock ]
-then
-  echo "This MicroK8s deployment is acting as a node in a cluster. Please use the microk8s enable on the master."
-  exit 0
-fi
+PARENT = Path(__file__).absolute().parent
+ADDONS = {sh.with_suffix('').name[7:] for sh in PARENT.glob('actions/enable.*.sh')}
 
-if echo "$*" | grep -q -- '--help'; then
-    LD_LIBRARY_PATH=$IN_SNAP_LD_LIBRARY_PATH ${SNAP}/usr/bin/python3 ${SNAP}/scripts/wrappers/enable.py $@
-    exit
-fi
 
-exit_if_stopped
-exit_if_no_permissions
+@click.command(context_settings={'ignore_unknown_options': True})
+@click.argument('addons', nargs=-1)
+def enable(addons):
+    check_clustering()
 
-result=1
-for addon in "$@"; do
-  # Making sure the cluster is up and running before each addon
-  action="$(addon_name $addon)"
-  arguments="$(addon_arguments $addon)"
-  # Check if the addon is already enabled, then skip.
-  if ${SNAP}/microk8s-status.wrapper --wait-ready --timeout 30 --addon $action | grep "enabled" >/dev/null; then
-    echo "Addon $action is already enabled."
-    result=0
-    continue
-  fi
-  
-  # If there is a script to execute for the $action run the script and ignore any yamls
-  if [ -f "${SNAP}/actions/enable.$action.sh" ]; then
-    "${SNAP}/actions/enable.$action.sh" "${arguments}"
-    enable_result="$?"
-    if [ "$enable_result" = "0" ]
-    then
-      result=0
-    else
-      echo "Failed to enable $action"
-      exit 1
-    fi
-  elif [ -f "${SNAP}/actions/$action.yaml" ]; then
-    echo "Enabling $action"
-    use_manifest $action apply
-    if [ "$use_manifest_result" = "0" ]
-    then
-      echo "$action enabled"
-      result=0
-    else
-      echo "Failed to enable $action"
-      exit 1
-    fi
-  else
-    echo "Nothing to do for $action"
-    exit 2
-  fi
-done
-exit $result
+    if all(a.split(':')[0] in ADDONS for a in addons) and len(addons) > 1:
+        for addon in addons:
+            addon, *args = addon.split(':')
+            run([PARENT / ('actions/enable.%s.sh' % addon)] + args.split(';'))
+    else:
+        addon, *args = addons[0].split(':')
```

Comment:
Seems like this would blow up with an IndexError if the user calls this without any args. Should probably have a check for an empty addons list and report a reasonable error.

Reply:
Added `required=True` which forces at least one argument to be passed, otherwise help text is displayed.